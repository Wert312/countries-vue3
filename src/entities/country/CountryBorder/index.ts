import CountryBorder from "@/entities/country/CountryBorder/ui/index.vue";

export default CountryBorder;
