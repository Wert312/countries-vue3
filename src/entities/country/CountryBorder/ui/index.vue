<template>
  <div class="borders">
    <div class="bordersWrapper">
      <span v-if="!borderCountries" class="noBorders">No Border Countries</span>
      <span v-if="borderCountries">Border Countries:</span>
            <router-link
              v-for="country in borderCountries"
              :key="country.name"
              :to="{
                name: 'CountryInfoPage',
                params: { country: country.name.official },
              }"
              class="borderLinks"
            >
              {{ country.name.common }}
            </router-link>
    </div>
  </div>
</template>

<script setup lang="ts">
import axios from "axios";
import {onMounted, PropType, ref} from "vue";
import type { Country } from "@/shared/models/country";

const props = defineProps({
  borderCodes: Array as PropType<string[]>,
});

let borderCountries = ref<Country[]>();

onMounted(() => {
  if (props.borderCodes) {
    let borderCountriesToString = props.borderCodes.join(",");
    return axios
      .get(
        `https://restcountries.com/v3.1/alpha?codes=${borderCountriesToString}`
      )
      .then((response) => (borderCountries.value = response.data));
  }
});

</script>

<style lang="scss">
.bordersWrapper {
  display: flex;
  flex-wrap: wrap;
  padding-top: 16px;
}

.borderLinks {
  margin: 0 5px 10px 0;
  display: table;
  align-self: center;
  padding: 2px 5px;
  background: var(--color-light);
  cursor: pointer;
  border-radius: 5px;
  text-decoration: none;
  color: var(--app-text);
  text-align: center;

  &:first-of-type {
    margin-left: 5px;
  }

  @include for-desktop {
    &:hover {
      box-shadow: var(--shadow-color);
      transition-property: box-shadow;
      transition-duration: 0.3s;
    }

    &:not(:hover) {
      transition-property: box-shadow;
      transition-duration: 0.3s;
    }
  }
}
</style>
