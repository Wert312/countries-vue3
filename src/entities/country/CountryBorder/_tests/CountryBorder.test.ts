import { mount } from "@vue/test-utils";
import { expect, it } from "vitest";
import CountryBorder from "../index";

describe("CountryBorder tests", () => {
  it("should render", () => {
    const wrapper = mount(CountryBorder);
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
