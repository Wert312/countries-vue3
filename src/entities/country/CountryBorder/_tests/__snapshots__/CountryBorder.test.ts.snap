// Vitest Snapshot v1

exports[`CountryBorder tests > should render 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="borders"
      >
        <div
          class="bordersWrapper"
        >
          <span
            class="noBorders"
          >
            No Border Countries
          </span>
          <!--v-if-->
          
          
        </div>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "borderCodes": undefined,
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "borderCountries": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": undefined,
              "_value": undefined,
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
            },
            "props": {
              "borderCodes": undefined,
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": [
            [Function],
          ],
          "next": null,
          "parent": [Circular],
          "props": {
            "borderCodes": undefined,
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "borderCodes": {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            [],
          ],
          "provides": {},
          "proxy": {
            "borderCodes": undefined,
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "borderCountries": undefined,
            "props": {
              "borderCodes": undefined,
            },
          },
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "No Border Countries",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": [],
                    "dynamicProps": null,
                    "el": <span
                      class="noBorders"
                    >
                      No Border Countries
                    </span>,
                    "key": 0,
                    "patchFlag": 0,
                    "props": {
                      "class": "noBorders",
                      "key": 0,
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "span",
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "v-if",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": [],
                    "dynamicProps": null,
                    "el": <!--v-if-->,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Comment),
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": ,
                    "appContext": null,
                    "children": [],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": [],
                    "dynamicProps": null,
                    "el": ,
                    "key": null,
                    "patchFlag": 128,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 16,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Fragment),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="bordersWrapper"
                >
                  <span
                    class="noBorders"
                  >
                    No Border Countries
                  </span>
                  <!--v-if-->
                  
                  
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "bordersWrapper",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "No Border Countries",
                "component": null,
                "dirs": null,
                "dynamicChildren": [],
                "dynamicProps": null,
                "el": <span
                  class="noBorders"
                >
                  No Border Countries
                </span>,
                "key": 0,
                "patchFlag": 0,
                "props": {
                  "class": "noBorders",
                  "key": 0,
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "span",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "v-if",
                "component": null,
                "dirs": null,
                "dynamicChildren": [],
                "dynamicProps": null,
                "el": <!--v-if-->,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Comment),
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [],
                "component": null,
                "dirs": null,
                "dynamicChildren": [],
                "dynamicProps": null,
                "el": ,
                "key": null,
                "patchFlag": 128,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Fragment),
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="borders"
            >
              <div
                class="bordersWrapper"
              >
                <span
                  class="noBorders"
                >
                  No Border Countries
                </span>
                <!--v-if-->
                
                
              </div>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "borders",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "C:/Users/averkhusha/WebstormProjects/countries-vue3/src/entities/country/CountryBorder/ui/index.vue",
            "__name": "index",
            "components": {},
            "props": {
              "borderCodes": [Function],
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="borders"
        >
          <div
            class="bordersWrapper"
          >
            <span
              class="noBorders"
            >
              No Border Countries
            </span>
            <!--v-if-->
            
            
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "C:/Users/averkhusha/WebstormProjects/countries-vue3/src/entities/country/CountryBorder/ui/index.vue",
          "__name": "index",
          "components": {},
          "props": {
            "borderCodes": [Function],
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="borders"
        >
          <div
            class="bordersWrapper"
          >
            <span
              class="noBorders"
            >
              No Border Countries
            </span>
            <!--v-if-->
            
            
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.38",
  },
  "__setProps": [Function],
  "componentVM": {
    "borderCodes": undefined,
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <div
    class="borders"
  >
    <div
      class="bordersWrapper"
    >
      <span
        class="noBorders"
      >
        No Border Countries
      </span>
      <!--v-if-->
      
      
    </div>
  </div>,
}
`;
